pytorch_lightning/utilities/cli.py

```diff
@@ -254,10 +269,10 @@ def __init__(
                 <https://jsonargparse.readthedocs.io/en/stable/#class-type-and-sub-classes>`_
                 of the given class.
         """
-        assert issubclass(trainer_class, Trainer)
-        assert issubclass(model_class, LightningModule)
+        assert issubclass(trainer_class, Trainer) if isinstance(trainer_class, type) else True
+        assert issubclass(model_class, LightningModule) if isinstance(model_class, type) else True
         if datamodule_class is not None:
-            assert issubclass(datamodule_class, LightningDataModule)
+            assert issubclass(datamodule_class, LightningDataModule) if isinstance(datamodule_class, type) else True
```

Comment:
I think i'd prefer removing these assertions since we already check in `add_lightning_class_args`

Reply:
Done :smiley: